File: src/runtime.c
#include <stdio.h>
#include <stdlib.h>

int runtime_getchar() {
    return getchar();
}

int runtime_putchar(int x) {
    return putchar(x);
}

int runtime_getint() {
    int x;
    scanf("%d", &x);
    return x;
}

int runtime_putint(int x) {
	return printf("%d", x);
}

int runtime_println(int x) {
    return printf("%d\n", x);
}

int runtime_exit(int x) {
	exit(x);
	return 0;
}

File: debug/runtime.go
package main

import "fmt"

func getchar() int {
	var n byte
	fmt.Scanf("%c", &n)
	return int(n)
}

func getint() int {
	var n int
	fmt.Scanf("%d", &n)
	return n
}

func putchar(n int) {
	fmt.Printf("%c", n)
}

func putint(n int) {
	fmt.Printf("%d", n)
}

func println(n int) {
	fmt.Printf("%d\n", n)
}
